### PR TITLE
mrtrix3: add version 3.0.3

### PR DESCRIPTION
**Pull request by bcdarwin**

### CODE REVIEW

File: var/spack/repos/builtin/packages/mrtrix3/package.py [hunk 1]
```diff
@@ -13,8 +13,10 @@ class Mrtrix3(Package):
        density."""
 
     homepage = "https://www.mrtrix.org/"
+    url      = "https://github.com/MRtrix3/mrtrix3/archive/refs/tags/3.0.3.tar.gz"
     git      = "https://github.com/MRtrix3/mrtrix3.git"
 
+    version('3.0.3', sha256='6ec7d5a567d8d7338e85575a74565189a26ec8971cbe8fb24a49befbc446542e', preferred=True)
     version('2017-09-25', commit='72aca89e3d38c9d9e0c47104d0fb5bd2cbdb536d')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/mrtrix3/package.py:20]: This `version` doesn't appear to be available (anymore?).  If it is, you could alternatively provide the appropriate `url` in the `version` directive.

```suggestion
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed sha256 for 3.0.3 but retention of `2017-09-25` is questionable.

**Comment by tldahlgren** [comment]
@adamjstewart @chuckatkins Would you care to comment on this PR and or my suggested changes?

File: var/spack/repos/builtin/packages/mrtrix3/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/mrtrix3/package.py:20]: If it's no longer available, we should deprecate it, not delete it. See https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/mrtrix3/package.py:20]: In that case, I will pretend I didn't check so we can reduce the PR backlog. 😉

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**